### PR TITLE
Adjusted Ask VA page to use accessible validation

### PR DESCRIPTION
**Pull request by samcoforma**
…instead of disabling submit button

## Summary

- removed the disabling of submit button
- added validation
- added accessible error response
- returned focus to checkbox when an error occurs

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/60177

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        | Screenshot 2023-07-26 at 9 18 55 AM |
| Desktop | 245254430-0dcf7eef-8ea8-452e-bdeb-87b21eeab9c2 | Screenshot 2023-07-26 at 9 18 30 AM |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/applications/claims-status/containers/AskVAPage.jsx [hunk 1]
```diff
@@ -63,10 +80,10 @@ class AskVAPage extends React.Component {
     } = this.props;
 
     const submitFunc = useLighthouse ? submit5103 : submitRequest;
-    const submitDisabled =
-      !this.state.submittedDocs ||
+    const canSubmit =
+      this.state.submittedDocs ||
       loadingDecisionRequest ||
-      decisionRequestError;
+      decisionRequestError !== null;
```
Comment (kjduensing) [inline src/applications/claims-status/containers/AskVAPage.jsx:86]: Do we need to check for other "falsey" values of `decisionRequestError`?

File: src/applications/claims-status/containers/AskVAPage.jsx [hunk 2]
```diff
@@ -109,23 +126,26 @@ class AskVAPage extends React.Component {
...
               </ul>
-              <div className="usa-alert usa-alert-info background-color-only claims-alert">
-                <Checkbox
-                  className="claims-alert-checkbox"
-                  checked={this.state.submittedDocs}
-                  onValueChange={update => this.setSubmittedDocs(update)}
-                  label="I have submitted all evidence that will support my claim and I’m not going to turn in any more information. I would like VA to make a decision on my claim based on the information already provided."
+              <div>
+                <VaCheckbox
+                  description=""
```
Comment (kjduensing) [inline src/applications/claims-status/containers/AskVAPage.jsx:131]: Just curious - should this be an empty string?

**Review by kjduensing — DISMISSED** [review]
LGTM 🦅

File: src/applications/claims-status/containers/AskVAPage.jsx [hunk 3]
```diff
@@ -109,23 +126,26 @@ class AskVAPage extends React.Component {
...
+                  error={this.state.submitErrorMessage}
+                  hint={null}
+                  label="I don't have any other documents to provide to support my claim. I'd like you to make a decision on my claim based on the evidence you have already."
+                  onVaChange={e => this.setSubmittedDocs(e.target.checked)}
+                  onBlur={function noRefCheck() {}}
```
Comment (jerekshoe) [inline src/applications/claims-status/containers/AskVAPage.jsx:136]: What does this do?

Comment (jerekshoe) [reply src/applications/claims-status/containers/AskVAPage.jsx:136]: I think this is a storybook thing and you don't need it

File: src/applications/claims-status/containers/AskVAPage.jsx (hunk 2 above)
Comment (jerekshoe) [reply src/applications/claims-status/containers/AskVAPage.jsx:131]: Probably don't need it

**Comment by jerekshoe** [comment]
Make sure you update the unit tests that are failing because of the changes made in this PR

**Comment by samcoforma** [comment]
@uxgary @uxgary Another note: make sure the "I have more evidence to submit" button takes the user back to the Files tab directly.

**Pull request closed without merging**